### PR TITLE
EGO Gifts 2: With a VENGEANCE

### PR DESCRIPTION
**Pull request by dsmith328**
## About The Pull Request
Adds a bunch more EGO Gifts, including code for non-implemented Abno's EGO Gifts.
Gives Blue Shepard a 50% weakness to Pale. He didn't have any weaknesses before.
White Night, due to now being an actual fucking beast, gives all those who survive from start to end of his rapture his gift and informs them that their fate shall not be a kind one.
White Lake has an EGO Gift that is not obtainable by normal means... she seems to only give it out to those she deems worthy.
Wearing Both Wings now also makes them not overlap.

## Why It's Good For The Game
More EGO Gifts are always nice to see. Rewarding those who manage to live through WN is also cool.

## Changelog
:cl:
add: EGO Gifts for a lot of Abnos. More credit to ButteredToast for spriting ALL OF THEM.
tweak: Blue Shepard's resistances. [With Permission]
add: EGO Gift reward from WN upon defeating him.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/ego_gifts.dm [hunk 1]
```diff
@@ -267,6 +402,26 @@
...
 
+/datum/ego_gifts/blossoming
+	name = "100 Paper Flowers"
+	icon_state = "blooming"
+	justice_bonus = 8
+	slot = SPECIAL
+
+/datum/ego_gifts/blossoming/Initialize(mob/living/carbon/human/user) // As a boost, undoes the debuff it applies to you
+	.=..()
+	user.physiology.red_mod /= 1.1
```
Comment (EgorDinamit) [inline code/modules/mob/living/carbon/human/ego_gifts.dm:413]: Use *= when added instead, so it's in line with the other buffs.

**Review by EgorDinamit — COMMENTED** [review]
As funny as it sounds, I'm not sure if giving a gift on WN supression is a good idea.
WN already rewards you for killing it with the most powerful weapon in game.

Additionally, you forgot to check if WN was killed by one sin.

File: code/modules/mob/living/carbon/human/ego_gifts.dm [hunk 2]
```diff
@@ -174,6 +224,63 @@
...
+	prudence_bonus = 4
+	slot = HELMET
+
+/datum/ego_gifts/desire
+	name = "Sanguine Desire"
+	icon_state = "desire"
+	fortitude_bonus = 4
+	slot = MOUTH_2
+
+/datum/ego_gifts/thosewhoknowthecrueltyofwinterandthearomaofroses
```
Comment (EgorDinamit) [inline code/modules/mob/living/carbon/human/ego_gifts.dm:246]: The hell is this name.

Comment (dsmith328) [reply code/modules/mob/living/carbon/human/ego_gifts.dm:246]: That is the ACCURATE NAME of the EGO GIFT. It is NOT Frost Splinter for WHATEVER GOD FORSAKEN REASON.
image

**Comment by dsmith328** [comment]
> As funny as it sounds, I'm not sure if giving a gift on WN supression is a good idea. WN already rewards you for killing it with the most powerful weapon in game.
> 
> Additionally, you forgot to check if WN was killed by one sin.

This is specifically for all those who LIVE through the suppression, so if you don't live through it you don't get the bonus. I did see Zacks suggestion and might implement that? I'll leave WN Gift stuff tweaking for another time in all honesty and just remove it for now

**Comment by EgorDinamit** [comment]
Honestly, I'd prefer if WN gave some other gift for suppressing it, the plague doctor blessing, for example.
Work gifts should remain work gifts, something you'd be surprised about(2% chance and all that).

File: code/modules/mob/living/simple_animal/abnormality/aleph/white_night.dm [hunk 3]
```diff
@@ -53,6 +53,9 @@ GLOBAL_LIST_EMPTY(apostles)
 	var/list/been_hit = list()
 	/// Currently spawned apostles by this mob
 	var/list/apostles = list()
+	/// List of Living People on Breach
+	var/list/heretics = list()
+	var/sinned = FALSE
```
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/aleph/white_night.dm:58]: Pointless variable. Just make one sin reset 'heretics' list on confession work.

Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/aleph/white_night.dm:58]: Pointless variable. Just make one sin reset 'heretics' list on confession work.

File: code/modules/mob/living/simple_animal/abnormality/aleph/white_night.dm [hunk 4]
```diff
@@ -73,6 +76,17 @@ GLOBAL_LIST_EMPTY(apostles)
...
 			revive_humans()
 
+/mob/living/simple_animal/hostile/abnormality/white_night/death(gibbed)
+	if (sinned)
+		return ..()
+	for(var/mob/living/carbon/human/heretic in heretics)
+		if(heretic.stat == DEAD || !heretic.ckey)
+			continue
+		heretic.Apply_Gift(new /datum/ego_gifts/blessing)
+		playsound(src, 'sound/abnormalities/whitenight/apostle_bell.ogg', 50)
```
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/aleph/white_night.dm:86]: This plays the sound at WN's location.
Additionally, you might want to use the local proc instead, so only the player hears it and not everyone around them.

File: code/modules/mob/living/simple_animal/abnormality/aleph/white_night.dm [hunk 5]
```diff
@@ -174,6 +188,8 @@ GLOBAL_LIST_EMPTY(apostles)
 	holy_revival_cooldown = world.time + holy_revival_cooldown_base
 	..()
 	for(var/mob/M in GLOB.player_list)
+		if(M.stat != DEAD && istype(M, /mob/living/carbon/human))
```
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/aleph/white_night.dm:191]: ishuman exists.

Comment (EgorDinamit) [reply code/modules/mob/living/simple_animal/abnormality/aleph/white_night.dm:191]: Might also want to check if this mob has a player.
So, check for ckey.

File: code/modules/mob/living/simple_animal/abnormality/aleph/white_night.dm [hunk 6]
```diff
@@ -73,6 +75,15 @@ GLOBAL_LIST_EMPTY(apostles)
 				G.forceMove(T)
 			revive_humans()
 
+/mob/living/simple_animal/hostile/abnormality/white_night/death(gibbed)
+	for(var/mob/living/carbon/human/heretic in heretics)
+		if(heretic.stat == DEAD || !heretic.ckey)
+			continue
+		heretic.Apply_Gift(new /datum/ego_gifts/blessing)
+		playsound_local(get_turf(heretic), 'sound/abnormalities/whitenight/apostle_bell.ogg', 50)
```
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/aleph/white_night.dm:83]:
```suggestion
		heretic.playsound_local(get_turf(heretic), 'sound/abnormalities/whitenight/apostle_bell.ogg', 50)
```

Comment (dsmith328) [reply code/modules/mob/living/simple_animal/abnormality/aleph/white_night.dm:83]: ...right

**Pull request merged**